### DISCUSSION TITLE
android/data unaccessable

### DISCUSSION THREAD

**Discussion by ghost** [osmandapp/OsmAnd#12982] [General]
# 🚀 feature request

### Description
As result of android development, recorded tracks are no longer accessible by an external application which used to send recorded tracks to my server.

### Describe the solution you'd like

would it be possible to pick a custom directory for the tracks directory outside of /storage/emulated/0/Android/data? (apart existing option to save tracks by month)

### Describe alternatives you've considered

Downgrading back to android 10 doesn't feel like a good solution. 
Root permission still doesn't allow accessing those files.
Script moving track files to another directory would be sufficient, but accessing those files can only be achieved via storage access framework as nd i don't (yet) know how to use it.

**Comment by vshcherb** [comment]
Changes are coming pretty soon, any update after November 2021 will have to target Android 11 and there are significant changes as we see on Android 11 for that.
1. Shared storage won't be available any more
2. External, Multiuser, Internal will still be available though for SDCard app performance will degrade for large number of maps
3. **Only MultiUser storage** will be visible in My Files / File Manager application

**Comment by ghost** [comment]
is any of that actionable? i think i have the most recent version available form fdroid (i bought it, but then decided to stop using google apps) and still cannot use the track files in external applications..
would it be possible to pick a custom directory for the tracks directory outside of /storage/emulated/0/Android/data?

**Reply by tommycrock** [reply]
I have been able to set up an accessible folder within /storage/emulated/0/download although I wasn't able to migrate and lost my settings. I've since imported my maps, tracks, etc.

**Reply by mariush444** [reply]
someone on telegram wrote

```
-------------------------------------
shared folder on Android 11 and osmand 4.1

it is possible to use shared folder 
but the only folder that can be used is
SD_card/Android/media/net.osmand.plus/

it works OK in my case
I have access to this folder via osmand and total commander

because there are to many problems with updating osmand I recommend fresh installation 
!!! before the copy or export profiles and setting is needed. !!!

If some one can confirm this at own phone then probably it will be useful for others
I don't know when this possibility will be close in the future
-------------------------------------
```

It works fine for me too.

**Reply by ghost** [reply]
thanks, this could be useful (as i can access the Android/media), but i have no idea on how to set this up, would you please find a moment to tell me how you forced osmand to save its stuff into that path?

**Reply by tommycrock** [reply]
https://docs.osmand.net/en/main@latest/osmand/personal/storage#data-storage-folder-android

**Comment by vinc17fr** [comment]
A current workaround is to install the "Save to File" application and share the track with it. But this works for only one track. OsmAnd should have its own feature and be able to make several tracks available at the same time, and possibly automatically when a track is finished.

**Comment by vinc17fr** [comment]
Related issues: #9857 and #5558.
